### DISCUSSION TITLE
Tune historical data in QPM

### DISCUSSION THREAD

**Discussion by trcrztw** [IRIS-Solutions-Team/IRIS-Toolbox#337] [Q&A] [answered]
Hi,
please help understand how to add historical tunes to variables in basic quarterly projection model. For example I have 4 basic equations for my home economy and ar(1) equations for foreign country. For foreign economy all trends are calculated using hpf, but for home economy - kalman filter is used. I want to add a negative shock to gdp potential for 2020q2 and then make forecasts. I tried some thoughts but to no effect. If I use fcast_plan for something earlier then t, I get nothing. For example I add negative shock to 2020q2 and make forecast starting 2020q3, there is no effect. If I use 2020q3 as negative shock to forecast 2020q3 it works.
Thank you!!!

**Comment by tosapola** [comment, marked as answer]
Hi,

You mean you don't feel so good with the potential GDP (say, ybar) from Kalman filter and want to lower it a bit for 2020Q2 which is in the past?

Make an additional measurement to the model as follow:
`ybar_ = ybar;`
and in your database, make another empty time series ybar_.

Before filtering, put the value you want to tune (the exact level of ybar, not the shock) in ybar_ above, for example:
`ybar_(qq(2020,2)) = 1456;`
and then apply Kalman filter, your ybar in 2020Q2 will be fixed at 1456.

And for the sake of smoothness, you may need to tune some more points around 2020Q2 too.

Best,
Tos

**Reply by trcrztw** [reply]
But I input only original, observed series (e.g. output level, cpi) into kalman filter, so setting ybar before filtering doesn't help as it is overwritten with kalman values.

**Reply by tosapola** [reply]
You get it right, just one more trick though. 

To prevent it from overwriting, that's why you have to set up an aditional "fake" measurement variable and equation. I presume you have ybar so I added an equation above.

And you need to input this "fake" data too when you do filtering. It's like you pretend that you observe it at some point of time. Normally we input 3-4 points we want to tune, not the whole time series.

**Reply by trcrztw** [reply]
Thank you, this helped!

**Comment by jaromir-benes** [comment]
I second Tos's suggestion. One detail though - if the process for potential GDP is highly persistent, then imposing one tune on the level at a particular time will quite likely generate a smooth path before and after - which I guess is not what the OP wanted. My guess is that the tune is supposed to create an abrupt break. In that case, you also need, for instance, impose another tune for the period immediately before 2020Q2.

Best
Jaromir

**Comment by tosapola** [comment]
Oh, I see. A sudden loss in potential output due to Covid19 - that's what I did to my model too.
Yes, maybe at least 3-4 points along 2020-2021 to impose to make a good picture.

**Reply by trcrztw** [reply]
Do you mind sharing how you made the tune please?

**Reply by tosapola** [reply]
I dont have some handy code in place.

Can you please share your model instead? No need to share the whole thing, maybe just the .model file and the part that you do filtering. I might be able to point out some lines of code to add.
But if you can send something runable that would be great.

Best,
Tos
